app/knowledge_panels.py

```diff
@@ -1,167 +1,184 @@
 from typing import Union
 from urllib.parse import urlencode
 
-from .models import HungerGameFilter, country_to_ISO_code, facet_plural
+from .models import country_to_ISO_code, facet_plural
 from .off import data_quality, hungergame, last_edit, wikidata_helper
 
 
-def hunger_game_kp(
-    hunger_game_filter: HungerGameFilter,
-    value: Union[str, None] = None,
-    country: Union[str, None] = None,
-):
-    query = {}
-    description = ""
-    if hunger_game_filter == "country":
-        country = value
-        hunger_game_filter = value = None
-    if country is not None:
-        query["country"] = country
-        description = country
-    if hunger_game_filter is not None:
-        query["type"] = f"{hunger_game_filter}"
-        description = f"{hunger_game_filter}"
-    if value is not None:
-        query["value_tag"] = value
-        description = f"{value} {hunger_game_filter}"
-    questions_url = "https://hunger.openfoodfacts.org/questions"
-    if query:
-        questions_url += f"?{urlencode(query)}"
-    t_description = hungergame()
-    description = f"{t_description} {description}"
-    html = f"<p><a href='{questions_url}'>{description}</a></p>\n"
-    return {
-        "hunger-game": {
-            "elements": [
+class KnowledgePanels:
+    def __init__(self, facet, value, sec_facet, sec_value, country):
+        self.facet = facet
+        self.value = value
+        self.sec_facet = sec_facet
+        self.sec_value = sec_value
+        self.country = country
```

Comment:
Is this coorect?

Reply:
Oh I didn't think of it, I talked about polymorphism as a comparison. But that's a good idea indeed, I like it :-)

Reply:
How to test it though?